src/EditorFeatures/Core/EncapsulateField/AbstractEncapsulateFieldCommandHandler.cs

```diff
@@ -95,11 +100,6 @@ public bool ExecuteCommand(EncapsulateFieldCommandArgs args, CommandExecutionCon
             {
                 await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-                // We are about to show a modal UI dialog so we should take over the command execution
-                // wait context. That means the command system won't attempt to show its own wait dialog 
-                // and also will take it into consideration when measuring command handling duration.
-                context.TakeOwnership();
```

Comment:
interesting.  what happens here?  shoul.d we explicitly dispose the context instead?

Reply:
We could add an extension for TakeOwnership that does a dispose... but does that present problems for double dispose?